docs/undergrad.yaml

```diff
@@ -9,7 +9,7 @@ Linear algebra:
     vector subspace: 'subspace'
     quotient space: 'submodule.quotient'
     sum of subspaces: 'submodule.complete_lattice'
-    direct sum: ''
+    direct sum: '' # this is supposed to be the internal direct sum
```

Comment:
Is the missing statement something like this?
```lean
def equiv_direct_sum
  (V : ι → submodule R M)
  (disjoint : ∀ i j (h : i ≠ j), disjoint (V i) (V j)) :
  ↥(⨆ i : ι, V i) ≃ₗ[R] (⨁ i, (V i)) := sorry
```

Reply:
`disjoint` is not enough: you need them to be "linearly independent": for any `f : Π₀ i, V i`, if the sum of vectors in `M` equals zero, then all `f i` are zeros. In other terms, the `inv_fun` of the equivalence is injective.

Reply:
A non-trivial statement here is a `tfae` about linear independent submodules. It should include at least:

* for any `f : Π₀ i, V i`, if the sum of vectors in `M` equals zero, then all `f i` are zeros;
* the natural map from `⨁ i, V i` is injective;
* for each `i`, `V i` is disjoint with `⨆ j ≠ i, V j`;
* for any two disjoint sets `s` and `t`, `⨆ i ∈ s, V i` is disjoint with `⨆ i ∈ t, V i`.

Note that we have some of these lemmas in `linear_algebra/pi` and `linear_algebra/std_basis` for the submodules `range (single i)`. I don't know whether we can reuse the new `tfae` here without creating an import loop.